src/test/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtServiceTest.java

```diff
@@ -0,0 +1,154 @@
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.cmc.claimstore.courtfinder.CourtFinderApi;
+import uk.gov.hmcts.cmc.claimstore.courtfinder.models.Court;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.legaladvisor.HearingCourt;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.legaladvisor.HearingCourtMapper;
+
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class PilotCourtServiceTest {
+
+    @Mock
+    private CourtFinderApi courtFinderApi;
+
+    @Mock
+    private HearingCourtMapper hearingCourtMapper;
+
+    private String csvPath = "/pilot-court/pilot-courts.csv";
+
+    private String csvPathSingle = "/pilot-court/pilot-courts-single.csv";
+
+    @Nested
+    @DisplayName("Init")
+    class InitTests {
+
+        @Test
+        void shouldThrowIllegalStateExceptionOnInvalidCSVPath() {
+            Assertions.assertThrows(IllegalStateException.class, () -> {
+                new PilotCourtService(
+                    "InvalidPath",
+                    courtFinderApi,
+                    hearingCourtMapper
+                ).init();
+            });
+        }
+
+        @Test
+        void shouldBuildListOfCourtsFromCSV() {
+            PilotCourtService pilotCourtService = new PilotCourtService(
+                csvPath,
+                courtFinderApi,
+                hearingCourtMapper
+            );
+
+            pilotCourtService.init();
+
+            Set<String> allPilotCourtIds = pilotCourtService.getAllPilotCourtIds();
+
+            assertEquals(4, allPilotCourtIds.size());
+            assertTrue(allPilotCourtIds.contains("EDMONTON"));
+            assertTrue(allPilotCourtIds.contains("MANCHESTER"));
+            assertTrue(allPilotCourtIds.contains("BIRMINGHAM"));
+            assertTrue(allPilotCourtIds.contains("CLERKENWELL"));
+        }
+    }
+
+    @Test
+    void shouldThrowRIllegalArgumentExceptionOnUnknownCourtId() {
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            PilotCourtService pilotCourtService = new PilotCourtService(
+                csvPathSingle,
+                courtFinderApi,
+                hearingCourtMapper
+            );
+            pilotCourtService.init();
+            pilotCourtService.getHearingCourt("UNKNOWN_ID");
+        });
+    }
+
+    @Test
+    void shouldReturnHearingCourt() {
+        PilotCourtService pilotCourtService = new PilotCourtService(
+            csvPathSingle,
+            courtFinderApi,
+            hearingCourtMapper
+        );
```

Comment:
This is common; could sit in a nested class's `@BeforeEach`

Reply:
Further stories expand this class and it is now nested